atomic_reactor/plugins/post_export_operator_manifests.py

```diff
@@ -78,79 +47,52 @@ def run(self):
 
         :return: str, path to operator manifests zip file
         """
-        if not self.should_run():
+        if not (
+                has_operator_bundle_manifest(self.workflow) or
+                has_operator_appregistry_manifest(self.workflow)
+        ):
+            self.log.info("Operator manifests label not set in Dockerfile. Skipping")
             return
 
-        # OSBS2 TBD this should go in the context dir
-        manifests_archive_dir = tempfile.mkdtemp()
-#        image = self.workflow.image
-        # OSBS2 TBD
-        # just use oc image extract to get image content
-#        container_dict = self.tasker.create_container(image, command=['/bin/bash'])
-#        container_id = container_dict['Id']
-#        try:
-#            bits, _ = self.tasker.get_archive(container_id,
-#                                              IMG_MANIFESTS_PATH)
-#        except APIError as ex:
-#            msg = ('Could not extract operator manifest files. '
-#                   'Is there a %s path in the image?' % (IMG_MANIFESTS_PATH))
-#            self.log.debug('Error while trying to extract %s from image: %s',
-#                           IMG_MANIFESTS_PATH, ex)
-#            self.log.error(msg)
-#            raise RuntimeError('%s %s' % (msg, ex)) from ex
-#
-#        except Exception as ex:
-#            raise RuntimeError('%s' % ex) from ex
-#
-#        finally:
-#            try:
-#                self.tasker.remove_container(container_id)
-#            except Exception as ex:
-#                self.log.warning('Failed to remove container %s: %s', container_id, ex)
-        bits = b''
-
-        with tempfile.NamedTemporaryFile() as extracted_file:
-            for chunk in bits:
-                extracted_file.write(chunk)
-            extracted_file.flush()
-            tar_archive = tarfile.TarFile(extracted_file.name)
-
-        tar_archive.extractall(manifests_archive_dir)
-        manifests_path = os.path.join(manifests_archive_dir, MANIFESTS_DIR_NAME)
+        platforms = get_platforms(self.workflow.data)
+        image = self.workflow.data.tag_conf.get_unique_images_with_platform(platforms[0])[0]
+        tmp_dir = tempfile.mkdtemp(dir=self.workflow.build_dir.platform_dir(platforms[0]).path)
```

Comment:
How about use `BuildDir.any_platform` consistently in this plugin?

Reply:
I was a little hesitant because I wasn't sure if we should rely on `BuildDir.any_platform` to return the directory for platform[0] only. I'll look into it and make sure it's consitant.

Reply:
IMO it doesn't matter which platform we use, `any_platform` should be OK

Reply:
updated